app/uk/gov/hmrc/gform/dms/DmsSubmissionAlgebra.scala

```diff
@@ -65,7 +67,28 @@ class DmsSubmissionService[F[_]](
     } yield envId
   }
 
+  override def submitPdfToDmsWithAttachments(html: String, attachments: List[Array[Byte]], metaData: DmsMetadata)(
+    implicit hc: HeaderCarrier): F[EnvelopeId] = {
+
+    val formTemplateId = FormTemplateId(metaData.dmsFormId)
+
+    for {
+      pdfBytes   <- pdfGenerator.generatePDFBytes(decode(html))
+      envelopeId <- fileUpload.createEnvelope(formTemplateId, LocalDateTime.now(clock).plusDays(formExpiryDays))
+      pdfDoc = documentLoader(pdfBytes)
+      pdfSummary = PdfSummary(pdfDoc.getNumberOfPages.toLong, pdfBytes)
+      _ = pdfDoc.close()
+      submission = DmsSubmissionService
+        .createSubmission(metaData, envelopeId, LocalDateTime.now(clock), Attachments.empty)
```

Comment:
I went through the code and I still can't see how the attachments are processed. All I can see is that it calculates the number of attachments. What about the content? How is that processed and sent down to DMS with the PDF?

Reply:
You need to send attachments into `file-upload-frontend`, that logic is not here...

Reply:
:man_facepalming: 

ok mate, let me check it out and see what I need to do - missing jigsaw in the puzzle.